src/Sylius/Bundle/CartBundle/Controller/CartItemController.php

```diff
@@ -61,15 +60,15 @@ public function addAction(Request $request)
             }
 
             $cart = $this->getCurrentCart();
-            $this->resolveCartItem($cart, $newResource);
+            $this->getCartModifier()->addToCart($cart, $newResource);
 
             $this->eventDispatcher->dispatchPostEvent(ResourceActions::CREATE, $configuration, $newResource);
 
             if (!$configuration->isHtmlRequest()) {
                 return $this->viewHandler->handle($configuration, View::create($newResource, Response::HTTP_CREATED));
             }
 
-            $this->getEventDispatcher()->dispatch(SyliusCartEvents::CART_CHANGE, new GenericEvent($cart));
+            $this->getEventDispatcher()->dispatch(SyliusCartEvents::ITEM_ADDED_TO_CART, new GenericEvent($cart));
```

Comment:
I think it was correct. We already dispatch order item `create` event and that should be enough, but we still need `cart_change` event to be dispatched on each cart modification (also add/remove something) at least for now.